compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -233,14 +233,17 @@ fn unroll_loop(
 /// The loop pre-header is the block that comes before the loop begins. Generally a header block
 /// is expected to have 2 predecessors: the pre-header and the final block of the loop which jumps
 /// back to the beginning.
-fn get_pre_header(cfg: &ControlFlowGraph, loop_: &Loop) -> BasicBlockId {
+fn get_pre_header(cfg: &ControlFlowGraph, loop_: &Loop) -> Result<BasicBlockId, CallStack> {
     let mut pre_header = cfg
         .predecessors(loop_.header)
         .filter(|predecessor| *predecessor != loop_.back_edge_start)
         .collect::<Vec<_>>();
 
-    assert_eq!(pre_header.len(), 1);
-    pre_header.remove(0)
+    if pre_header.len() == 1 {
+        Ok(pre_header.remove(0))
+    } else {
+        Err(CallStack::new())
```

Comment:
It would be helpful to add some comments to this function. For example it's not obvious to me what the error represents.

Reply:
Ah, good point. But the code was already returning `Callstack::new` in some other places so I just followed what was done here without really understanding why that is returned. I'll briefly take a look at this to see why a Callstack is returned and try to change it or add a comment.

Reply:
It seems all RuntimeErrors have must have a call_stack, and here a dummy one is used, just to comply to that requirement (it's a `call_stack` method on `RuntimeError`). The call stack is not used when turning that error into a message... for now I'll leave it like this because it's unrelated to this PR changes, if that's okay with you.

Reply:
Yes, I also had a look and saw that it can only be produced when an unexpected `TerminatorInstruction` is found, which carries a location. Based on the docs of `unroll_loop` this wasn't always returned. The only comment I would add here is that when `pre_header.len() > 1` is I believe when `break` or `continue` is used, in which case we cannot inline and we will discard this entire attempt as an error.